macros/get_metric_sql.sql

```diff
@@ -4,19 +4,55 @@
       - allow start/end dates on metrics. Maybe special-case "today"?
       - allow passing in a seed with targets for a metric's value
 */
-{%- macro get_metric_sql(metric_list, grain, dimensions, secondary_calculations, start_date, end_date, where) %}
+{%- macro get_metric_sql(metric_list, grain, dimensions, secondary_calculations, start_date, end_date, where, initiated_by,metric_definition=None) %}
 
+{# ############
+VALIDATION AROUND METRIC VS CALCULATE VS DEVELOP
+############ #}
+
+{% if initiated_by == 'calculate' %}
+    {% set is_calculate_macro = true %}
+    {% set is_develop_macro = false %}
+{% elif initiated_by == 'develop' %}
+    {% set is_calculate_macro = false %}
+    {% set is_develop_macro = true %}
+{% endif %}
+
+{# ############
+METRIC ATTRIBUTES TO VARIABLES
+We do this so that we can support metric attributes coming in from other formats
+like develop. This is easier than altering all of the downstream logic to have flags
+for calculate vs develop
+############ #}
+{% if is_develop_macro %}
+    {% set metric_name = metric_definition["name"]%}
+    {% set metric_type = metric_definition["type"]%}
+    {% set metric_sql = metric_definition["sql"]%}
+    {% set metric_timestamp = metric_definition["timestamp"]%}
+    {% set metric_time_grains = metric_definition["time_grains"]%}
+    {% set metric_dimensions = metric_definition["dimensions"]%}
+    {% set metric_filters = metric_definition["filters"]%}
+    {% set metric_base_model = metric_definition["model"].replace('"','\'').split('\'')[1]  %}
+    {% set metric_model = metrics.get_model_relation(metric_base_model if execute else "") %}
+{% endif %}
+ 
 {# ############
 VARIABLE SETTING ROUND 1: List Vs Single Metric!
 ############ #}
 
-{% if metric_list is not iterable %}
+{% if is_calculate_macro and metric_list is not iterable %}
     {% set metric_list = [metric_list] %}
+{% elif is_develop_macro %}
+    {% set metric_list = [metric_definition["name"]] %}
 {% endif %}
 
 {# We are creating the metric tree here - this includes all the leafs (first level parents)
 , the expression metrics, and the full combination of them both #}
-{%- set metric_tree = metrics.get_metric_tree(metric_list) %}
+{% if is_calculate_macro %}
+    {% set metric_tree = metrics.get_metric_tree(metric_list) %}
+{% elif is_develop_macro %}
+    {% set metric_tree = metrics.get_faux_metric_tree(metric_list) %}
```

Comment:
This feels... ick. Again because I don't really like that this macro has to know about the two different development modes. 

An idea I just had but haven't validated: could the metric tree be passed in to get_metric_sql instead? I.e. in the `calculate` and `develop` metrics (which I assume are what call this) they are each responsible for generating a correctly-shaped tree, however they need to go about it (meaning the `faux tree` approach is OK for develop).

I would feel much better about that.

Reply:
I am .... actually totally on board with this idea. Making the update now